lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -106,6 +113,22 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
     }
   }
 
+  void killAfterTimeout() {
+      ::sleep(10);
+
+      if (!done) {
+        sendSignalToProcesses(SIGKILL);
+      }
+  }
+
+  void sendSignalToProcesses(int signal) {
+      std::unique_lock<std::mutex> lock(spawnedProcessesMutex);
+
+      for (pid_t pid: spawnedProcesses) {
+          ::kill(-pid, signal);
```

Comment:
We should have a comment here that points out we kill the entire process group, and that this is dependent on us spawning each subprocess into its own process group (and also reference this fact in the code to spawn).